Public/Src/Engine/Scheduler/Distribution/Worker.cs

```diff
@@ -421,27 +421,54 @@ internal bool TryAcquire(RunnablePip runnablePip, out WorkerResource? limitingRe
                 loadFactor = 1;
             }
 
-            if (AcquiredProcessSlots >= (EffectiveTotalProcessSlots * loadFactor))
+            var processRunnablePip = runnablePip as ProcessRunnablePip;
+            var acquiredSlots = m_acquiredProcessSlots;

```

Comment:
Why don't use use the `AcquiredProcessSlots` property instead of reading the fields directly?  The property will read it using `Volatile.Read`, which might in some cases give you a more up-to-date value

Reply:
Ended up going back to the old format because it turns out all of this function is being protected by a mutex on the calling side.

---
In reply to: [271105791](https://github.com/Microsoft/BuildXL/pull/67#discussion_r271105791) [](ancestors = 271105791)